### ISSUE TITLE
the latest 4.0.x version of the configuration MasterSlaveRuleConfiguration properly

### ISSUE THREAD

**Issue by lonyee1989** [apache/incubator-shardingsphere#2846]
The latest 4.0.x version of the configuration MasterSlaveRuleConfiguration properly

The hard solution by lonyee
[https://www.jianshu.com/p/30175084c918](https://www.jianshu.com/p/30175084c918)

It is best to modify the write related config class.

**Comment by lonyee1989** [comment]
for #2846 4.0.x version of the configuration MasterSlaveRuleConfiguration properly

**Comment by terrymanu** [comment]
We should update document later, but a full parameters' constructor is necessary.

I will open a new issue on document repo

**Comment by lonyee1989** [comment]
OK... I think custom yaml configuration should be supported, and does configuration validation work in the official yaml without parametric constructors?

like this? @terrymanu 
```
masterSlaveRules:
      ms_ds0:
        masterDataSourceName: ds0
        slaveDataSourceNames:
          - ds0_slave0
          - ds0_slave1
        loadBalanceAlgorithmType: ROUND_ROBIN
      ms_ds1:
        masterDataSourceName: ds1
        slaveDataSourceNames: 
          - ds1_slave0
          - ds1_slave1
        loadBalanceAlgorithmType: ROUND_ROBIN
```

**Comment by tristaZero** [comment]
Hi, @lonyee1989 Thanks for your issue , can you fix this problem and raise a pr for [Sharding Sphere document](https://github.com/apache/incubator-shardingsphere-doc)? Thanks a lot

**Comment by lonyee1989** [comment]
@tristaZero ok, I'll try it.

**Comment by tristaZero** [comment]
Hi，we plan to release new version tomorrow, so can you fix this bug today or tomorrow?:)
@lonyee1989

**Comment by lonyee1989** [comment]
I'm going to do it tomorrow and try to finish it within this week. @tristaZero

**Comment by tristaZero** [comment]
Thanks for your comprehension, please raise pr to [doc](https://github.com/apache/incubator-shardingsphere-doc).
@lonyee1989 (ﾟ▽ﾟ)/

**Comment by lonyee1989** [comment]
@tristaZero I have submitted the update of the document.   [shardingsphere-doc#224](https://github.com/apache/incubator-shardingsphere-doc/pull/224)

**Comment by tristaZero** [comment]
@lonyee1989 Hi, you checked and changed a lot of content of document, thinks a lot.
But actually, we change the api and document of encryption in the latest verssion i.e, 4.0.0.RC2 , so the description of encryption in document is right. 
`Can you delete the changes of encryption in your pr and just keep the changes about masterSlaveConfig?`
What's more, masterSlaveConfig is used a lot in document, `can you find them and change them togather?`
Thanks a lot, our new version needs your contribution :).

**Comment by lonyee1989** [comment]
aha, I have fixed it. [shardingsphere-doc#227](https://github.com/apache/incubator-shardingsphere-doc/pull/227) .  :) waiting for the version 4.0.0.RC2

**Comment by lonyee1989** [comment]
@tristaZero aha, I have fixed it. [shardingsphere-doc#227](https://github.com/apache/incubator-shardingsphere-doc/pull/227)  
:) waiting for the version 4.0.0.RC2

**Comment by tristaZero** [comment]
@lonyee1989 merged it , good job! 4.0.0.RC2 is coming soon. Thanks for your contribution:)

**Comment by tristaZero** [comment]
Thanks waiting for you, please raise pr to [doc]()
